### PR TITLE
Fix to work with spec compliant CAN sims.

### PR DESCRIPTION
**Pull request by brettle**
Rename SimDeviceEncoderMediator.java to CANEncoderMediator.java.

Rename SimDeviceMotorMediator.java to CANMotorMediator.java.

Add DutyCycleMediator.java to handle SparkMaxAbsoluteEncoder and CANCoder devices.

Add AnalogInputMediator.java to handle SparkMaxAnalogSensor devices.

Currently passes system test but that doesn't test any of the new functionality.

### CODE REVIEW

**Comment by brettle** [comment]
Includes PR #50 and PR #49.

File: plugin/controller/src/main/java/org/team199/deepbluesim/SimRegisterer.java [hunk 1]
```diff
@@ -128,22 +134,29 @@ public static void connectEncoder(PositionSensor device, Supervisor robot) {
...
                 return;
             }
 
             String[] motorNameParts = motorName.split("_");
             int motorId = Integer.parseInt(motorNameParts[3]);
 
-            String simDeviceName = "SparkMax[" + motorId + "]_" + node.getTypeName().substring("SparkMax".length());
-
-            new SimDeviceEncoderMediator(device, new SimDeviceSim(simDeviceName), isOnMotorShaft, isAbsolute, absoluteOffsetDeg, isInverted, countsPerRevolution, gearing);
+            if (node.getTypeName().equals("SparkMaxAlternateEncoder")) {
```
Comment (CoolSpy3) [inline plugin/controller/src/main/java/org/team199/deepbluesim/SimRegisterer.java:150]: Given that we're adding support for SparkFlexes, it probably makes sense to rename a lot of these. However, that might be better suited for another pr.

File: plugin/controller/src/main/java/org/team199/deepbluesim/mediators/AnalogInputMediator.java
Comment (CoolSpy3) [inline plugin/controller/src/main/java/org/team199/deepbluesim/mediators/AnalogInputMediator.java:1]: Could we rename this to `AnalogInputEncoderMediator` to reflect the fact that it mediates encoders and not generic analog sources?

File: plugin/controller/src/main/java/org/team199/deepbluesim/mediators/CANMotorMediator.java [hunk 2]
```diff
@@ -39,19 +41,19 @@ public class SimDeviceMotorMediator implements Runnable {
...
-    public SimDeviceMotorMediator(Motor motor, SimDeviceSim simDevice, DCMotor motorConstants, double gearing, boolean inverted, Collection<ScopedObject<?>> callbackStore) {
+    public CANMotorMediator(Motor motor, CANMotorSim simDevice, DCMotor motorConstants, double gearing, boolean inverted, Collection<ScopedObject<?>> callbackStore) {
         this.motor = motor;
         motorDevice = simDevice;
         this.motorConstants = motorConstants;
         this.gearing = gearing;
         this.inverted = inverted;
 
-        if(motor.getName().startsWith("DBSim_Motor_Spark Max")) {
+        if(motor.getName().startsWith("DBSim_Motor_Spark ")) {
```
Comment (CoolSpy3) [inline plugin/controller/src/main/java/org/team199/deepbluesim/mediators/CANMotorMediator.java:51]: To be consistent with the check in `SimRegisterer.java`
```suggestion
        if(motor.getName().startsWith("DBSim_Motor_Spark")) {
```

File: plugin/controller/src/main/java/org/team199/deepbluesim/mediators/CANMotorMediator.java [hunk 3]
```diff
@@ -96,7 +98,7 @@ public void run() {
         double currentDraw = motorConstants.getCurrent(velocity, currentOutput * motorConstants.nominalVoltageVolts);
         motor.setAvailableTorque(motorConstants.getTorque(currentDraw) * gearing);
 
-        motorDevice.set("Current Draw", currentDraw);
+        motorDevice.setMotorcurrent(currentDraw);
     }
```
Comment (CoolSpy3) [inline plugin/controller/src/main/java/org/team199/deepbluesim/mediators/CANMotorMediator.java:102]: To ensure spec compliance, we should also send `busVoltage = 12`.

**Review by CoolSpy3 — CHANGES_REQUESTED** [review]

File: plugin/controller/src/main/java/org/team199/deepbluesim/mediators/CANMotorMediator.java (hunk 3 above)
Comment (brettle) [reply plugin/controller/src/main/java/org/team199/deepbluesim/mediators/CANMotorMediator.java:103]: Fixed now.

**Review by CoolSpy3 — APPROVED** [review]

**Pull request merged**